src/common/module/yoethf.F90

```diff
@@ -72,9 +72,9 @@ MODULE YOETHF
 !$acc   r5alvcp, r5alscp, ralvdcp, ralsdcp, ralfdcp, rtwat, rtice, rticecu, &
 !$acc   rtwat_rtice_r, rtwat_rticecu_r, rkoop1, rkoop2)
 
-!$omp declare target(r2es, r3les, r3ies, r4les, r4ies, r5les, r5ies, &
-!$omp   r5alvcp, r5alscp, ralvdcp, ralsdcp, ralfdcp, rtwat, rtice, rticecu, &
-!$omp   rtwat_rtice_r, rtwat_rticecu_r, rkoop1, rkoop2)
```

Comment:
It just didn't work with line continuation, not sure why.

Reply:
For me the original version works on Mac, so would argue against changing this here

Reply:
Just for the record, I have reverted the change locally and tested with gfortran 12.2 from homebrew:
```
carl:napz-apple nabr$ git diff
diff --git a/src/common/module/yoethf.F90 b/src/common/module/yoethf.F90
index c3b5625..ff5d093 100644
--- a/src/common/module/yoethf.F90
+++ b/src/common/module/yoethf.F90
@@ -72,9 +72,9 @@ REAL(KIND=JPRB) :: RKOOP2
 !$acc   r5alvcp, r5alscp, ralvdcp, ralsdcp, ralfdcp, rtwat, rtice, rticecu, &
 !$acc   rtwat_rtice_r, rtwat_rticecu_r, rkoop1, rkoop2)

-!$omp declare target(r2es, r3les, r3ies, r4les, r4ies, r5les, r5ies)
-!$omp declare target(  r5alvcp, r5alscp, ralvdcp, ralsdcp, ralfdcp, rtwat, rtice, rticecu)
-!$omp declare target(  rtwat_rtice_r, rtwat_rticecu_r, rkoop1, rkoop2)
+!$omp declare target(r2es, r3les, r3ies, r4les, r4ies, r5les, r5ies, &
+!$omp   r5alvcp, r5alscp, ralvdcp, ralsdcp, ralfdcp, rtwat, rtice, rticecu, &
+!$omp   rtwat_rtice_r, rtwat_rticecu_r, rkoop1, rkoop2)

 !       ----------------------------------------------------------------

carl:napz-apple nabr$ gfortran --version
GNU Fortran (Homebrew GCC 12.2.0) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Reply:
For me it doesn't work, exactly the same version of gcc. Do your compiler flags include both -fopenacc and -fopenmp, as in the isolated example below?  :
Zbigniews-MacBook-Pro:common piotrows$ cd /Users/piotrows/dwarf-p-cloudsc-test/build/cloudsc-dwarf/src/common && /opt/homebrew/bin/gfortran-12 -fopenacc -fopenmp -c /Users/piotrows/dwarf-p-cloudsc-test/source/cloudsc-dwarf/src/common/module/yoethf.F90 -o CMakeFiles/cloudsc-common-lib.dir/module/yoethf.F90.o
/Users/piotrows/dwarf-p-cloudsc-test/source/cloudsc-dwarf/src/common/module/yoethf.F90:77:7:

   77 | !$omp  r5alvcp, r5alscp, ralvdcp, ralsdcp, ralfdcp, rtwat, rtice, rticecu, &
      |       1
Error: Wrong OpenACC continuation at (1): expected !$ACC, got !$OMP
/Users/piotrows/dwarf-p-cloudsc-test/source/cloudsc-dwarf/src/common/module/yoethf.F90:76:21:

   76 | !$omp declare target(r2es, r3les, r3ies, r4les, r4ies, r5les, r5ies, &
      |                     1
Error: Syntax error in !$OMP DECLARE TARGET list at (1)
/Users/piotrows/dwarf-p-cloudsc-test/source/cloudsc-dwarf/src/common/module/yoethf.F90:77:8:

   77 | !$omp  r5alvcp, r5alscp, ralvdcp, ralsdcp, ralfdcp, rtwat, rtice, rticecu, &
      |        1
Error: Unclassifiable OpenMP directive at (1)
/Users/piotrows/dwarf-p-cloudsc-test/source/cloudsc-dwarf/src/common/module/yoethf.F90:78:7:

   78 | !$omp  rtwat_rtice_r, rtwat_rticecu_r, rkoop1, rkoop2)
      |       1
Error: Wrong OpenACC continuation at (1): expected !$ACC, got !$OMP
/Users/piotrows/dwarf-p-cloudsc-test/source/cloudsc-dwarf/src/common/module/yoethf.F90:78:8:

   78 | !$omp  rtwat_rtice_r, rtwat_rticecu_r, rkoop1, rkoop2)
      |        1
Error: Unclassifiable OpenMP directive at (1)

Reply:
Indeed, I hadn't OpenACC activated as it doesn't make any sense on Mac without a corresponding offload target implementation. With both activated I'm seeing the same. Is the issue also present on x86_64 with gfortran 12?

Reply:
Interestingly, on the latest Mint with GCC 11 I get the following error when building:
[ 68%] Linking Fortran shared library ../../lib/libcloudsc-common-lib.so
mkoffload: fatal error: either '-fopenacc' or '-fopenmp' must be set

In turn, with GCC 12 I get the same error as on Mac:
 75%] Building Fortran object cloudsc-dwarf/src/common/CMakeFiles/cloudsc-common-lib.dir/module/yoethf.F90.o
/home/piotrows/dwarf-p-cloudsc/source/cloudsc-dwarf/src/common/module/yoethf.F90:76:7:

   76 | !$omp   r5alvcp, r5alscp, ralvdcp, ralsdcp, ralfdcp, rtwat, rtice, rticecu, &
      |       1
Error: Wrong OpenACC continuation at (1): expected !$ACC, got !$OMP
/home/piotrows/dwarf-p-cloudsc/source/cloudsc-dwarf/src/common/module/yoethf.F90:75:21:

   75 | !$omp declare target(r2es, r3les, r3ies, r4les, r4ies, r5les, r5ies, &
      |                     1
Error: Syntax error in !$OMP DECLARE TARGET list at (1)
/home/piotrows/dwarf-p-cloudsc/source/cloudsc-dwarf/src/common/module/yoethf.F90:76:9:

   76 | !$omp   r5alvcp, r5alscp, ralvdcp, ralsdcp, ralfdcp, rtwat, rtice, rticecu, &
      |         1
Error: Unclassifiable OpenMP directive at (1)
/home/piotrows/dwarf-p-cloudsc/source/cloudsc-dwarf/src/common/module/yoethf.F90:77:7:

   77 | !$omp   rtwat_rtice_r, rtwat_rticecu_r, rkoop1, rkoop2)
      |       1
Error: Wrong OpenACC continuation at (1): expected !$ACC, got !$OMP
/home/piotrows/dwarf-p-cloudsc/source/cloudsc-dwarf/src/common/module/yoethf.F90:77:9:

   77 | !$omp   rtwat_rtice_r, rtwat_rticecu_r, rkoop1, rkoop2)
      |         1

Reply:
By the way, main (which I cloned at first by accident) does not build at all on the newest Mint. Time to merge develop into main ? 
Error msg: 
 38%] Building Fortran object cloudsc-dwarf/src/prototype1/CMakeFiles/dwarf-P-cloudMicrophysics-IFSScheme-libs.dir/support/yomct0.F90.o
f951: Fatal Error: Cannot rename module file ‘module/hdf5_file_mod.mod0’ to ‘module/hdf5_file_mod.mod’: No such file or directory
compilation terminated.
make[2]: *** [cloudsc-dwarf/src/common/CMakeFiles/cloudsc-common-lib-static.dir/build.make:310: cloudsc-dwarf/src/common/CMakeFiles/cloudsc-common-lib-static.dir/module/hdf5_file_mod.F90.o] Error 1

Reply:
Thanks for confirming on Linux! I'm OK with the change.

The GCC 11 error sounds like some flag(s) have not been propagated to the linking stage.

Reply:
A new CLOUDSC release is already planned

Reply:
@mlange05 before we move on: is it at all proper for the CLOUDSC to employ -fopenmp and -fopenacc flags together ?

Reply:
@reuterbal it looks to me that the gcc11 offloader didn't like the setup with -fopenacc -fopenmp. However,  with gcc-12 it is better and I can confirm that with the omp fix all 7 tests are successfully passed on my i7-6700HQ (ESCAPE-purchased ;D) laptop with NVIDIA 960M.

Reply:
The OpenACC/OpenMP interoperability is something claimed by most compiler vendors but with varying maturity. It works fine on NVHPC for example, it requires some tricks with the right order of flags on Cray, and so on.
Offloading with GCC is nothing we've looked at extensively, so not surprised that we see some hiccups there. 
While there would be means and ways of making this more robust, in the interest of keeping the overall setup simple and easier to use for exploration, I'd argue to keep it as is, and build separately with only OpenACC or OpenMP enabled if necessary.

Reply:
> I'd argue to keep it as is, and build separately with only OpenACC or OpenMP enabled if necessary.
Agree, but it seems that if the offload library is installed, then the cloudsc  builder defaults to include both flags together. Since it is beyond the scope of this PR, perhaps we should delegate to a further development (silly enough, I can't find how to submit a feature request/bug report to the cloudsc repo).

Reply:
Yes, but `--cmake=ENABLE_OMP=OFF` or `--cmake=ENABLE_ACC=OFF` is sufficient to switch one of them off.

Reply:
Yes, but then we need either include these three declares instead of ampersands to mitigate the -seemingly- compiler bug, or we need to change the Readme to inform the reader to use the switches you provided above. As a lazy person, I would be in favour of the former solution, as it is standard-compliant and do not put any burden on the end-user. However, I may miss an important issue, perhaps @mlange may provide an insight as well here.

Reply:
Like I said above, I'm ok with the change to separate declarations. I suggested retaining the original version only because I couldn't reproduce the issue in the first place.